src/lib.rs

```diff
@@ -532,7 +532,11 @@ pub struct BasicDataFormatDescriptor<'data> {
 
 impl<'data> BasicDataFormatDescriptor<'data> {
     pub fn parse(bytes: &'data [u8]) -> Result<Self, ParseError> {
-        let header = BasicDataFormatDescriptorHeader::from_bytes(bytes)?;
+        if bytes.len() < BasicDataFormatDescriptorHeader::LENGTH {
+            return Err(ParseError::UnexpectedEnd);
+        }
+        let header_data = bytes[0..BasicDataFormatDescriptorHeader::LENGTH].try_into().unwrap();
```

Comment:
Style nit: replace if + slice with `get(0..etc).ok_or(ParseError::UnexpectedEnd)?`.

Reply:
I copied this code verbatim from https://github.com/BVE-Reborn/ktx2/blob/9c534c4dd44003ede8006fdbca7aa3de9257e47f/src/lib.rs#L48-L51 -- should we update both, just this one, or keep them the same?

Reply:
I think this is a better pattern, don't you? Let's use it wherever applicable, though don't feel obliged to update code you're not touching.

Reply:
I agree! I'll update the code I'm touching.